OpenTrons/opentrons-api
# fix(app): Handle Unsafe Move to Plunger during Drop-Tip

Closes [EXEC-186](https://opentrons.atlassian.net/browse/EXEC-186)


# Overview

If the gantry is not homed and a powercycle occurs, drop-tip wizard cannot proceed with flows. An error is raised during the flow, and ultimately a home command is dispatched that has the side effect of potentially aspirating liquid into the pipette, damaging it. We need to special case home errors to prevent this. See ticket for designs.

The primary functional difference is now that any time an error occurs, exiting the wizard via the header should not home the gantry. Homing as a result of an error should only occur when the "Confirm removal and home" button is presented and clicked.

There's some minor refactor work done in this PR as well. More refactoring is definitely needed, but that should be handled in a separate PR (and take into account the wizard's use in the new error recovery flows). 

Note that I'm doing something a bit different for testing purposes, as a lot of our large wizard flows lack testing and are hard to appropriately approach with one big DOM driven integration test. Let's create some local utility functions and unit test those. Let me know what ya'll think.

### ODD
<img width="665" alt="Screenshot 2024-04-15 at 1 42 25 PM" src="https://github.com/Opentrons/opentrons/assets/64858653/f86c5d56-e093-40db-a130-44f9d43ec77a">

### Desktop
<img width="1018" alt="Screenshot 2024-04-15 at 1 00 33 PM" src="https://github.com/Opentrons/opentrons/assets/64858653/af2d17f3-221c-44ad-b3e1-6bbee4a8888d">



# Test Plan
- Do something that causes the gantry not to be homed. Power cycle the robot.
- Launch drop tip wizard, and proceed through the flow until the `MustHomeError` occurs.
- Confirm error recovery functionality works as intended based on designs.


# Changelog
- Fixed unsafe move plunger commands that may occur drop tip wizard. 





# Risk assessment
low



[EXEC-186]: https://opentrons.atlassian.net/browse/EXEC-186?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
